airbyte-integrations/connectors/source-postgres/src/main/kotlin/io/airbyte/integrations/source/postgres/operations/PostgresSourceSelectQueryGenerator.kt

```diff
@@ -90,7 +90,7 @@ class PostgresSourceSelectQueryGenerator : SelectQueryGenerator {
                     if (sampleRateInv == 1L) {
                         ""
                     } else {
-                        " TABLESAMPLE SYSTEM(${sampleRatePercentage.toPlainString()}) REPEATABLE(2472983)"
+                        " TABLESAMPLE SYSTEM(GREATEST(${sampleRatePercentage.toPlainString()}, 0.001))"
```

Comment:
Note: Postgres' `TABLESAMPLE SYSTEM` is returning quickly but instead of distributed sample it returned clusters of rows on the same page (addresses in postgres are `(page, tuple)`.
Similar to what @burakku is experiencing with MSSQL.

Setting a floor value of 0.001 will get more rows from which the 1024 will be chosen, increasing the distribution.
On 1TB table the sample is 60 times better, and only adding a few more seconds.